clang/lib/Driver/SanitizerArgs.cpp

```diff
@@ -1137,6 +1137,20 @@ void SanitizerArgs::addArgs(const ToolChain &TC, const llvm::opt::ArgList &Args,
       return;
     GPUSanitize = true;
   }
+  // SPIR sanitizer support is experimental and will pass a fixed set of flags
+  if (TC.getTriple().isSPIR()) {
+    if (Sanitizers.has(SanitizerKind::Address)) {
+      CmdArgs.push_back("-fsanitize=address");
+      CmdArgs.push_back("-fsanitize-address-use-after-return=never");
+      CmdArgs.push_back("-fsanitize-address-outline-instrumentation");
+      CmdArgs.push_back("-fno-sanitize-address-use-after-scope");
+      CmdArgs.push_back("-mllvm");
+      CmdArgs.push_back("-asan-stack=0");
+      CmdArgs.push_back("-mllvm");
+      CmdArgs.push_back("-asan-globals=1");
```

Comment:
Can a test be added to verify these options being passed for the device compile?

Reply:
done